src/components/form/form.ts

```diff
@@ -0,0 +1,31 @@
+import {Block} from "../block/block";
+import {blockProperty} from "../block/types";
+import {Input} from "../Input/input";
+import {ContaineredInput} from "../containeredInput/containeredInput";
+
+
+export class Form extends Block {
+  
+  constructor(props: blockProperty, template?: string) {
+    super("form", props, template);
+  }
+  
+  validate(): boolean {
+    const forValidate = Object.values(this.props).filter((item: any) => item instanceof Input
+      || item instanceof ContaineredInput);
+    let isNotValid: boolean = false;
+    for (let i = 0; i < forValidate.length; i++) {
+      isNotValid = !!forValidate[i].validate() || isNotValid;
```

Comment:
Удобней будет возвращать объект/массив и деструктуризовать, что требуется
```suggestion
      const [isCurrentFieldValid, message] = forValidate[i].validate();
      isNotValid = isCurrentFieldValid || isNotValid;
```
Ненужное можно опускать
```
const [isCurrentFieldValid, ] = forValidate[i].validate();
const [, message] = forValidate[i].validate();
```